src/Biryuchkov/Task5/Sphere.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Sphere.h"
+#include <utility>
+#include <cmath>
+
+
+Sphere::Sphere(const QVector3D &c,const float r, Material m)
+        : center(c), radius(r), material(std::move(m)){}
+
+bool Sphere::ray_intersect(const Ray &ray, float &t) const {
+    QVector3D L =  center - ray.origin;
+    auto tca = QVector3D::dotProduct(L,ray.direction);

```

Comment:
```suggestion
    const auto tca = QVector3D::dotProduct(L,ray.direction);

```

делайте все константным по умолчанию, и хорошо бы называть переменные более понятным образом